pkg/cloudprovider/aws/capacity.go

```diff
@@ -35,34 +49,57 @@ type Capacity struct {
 	instanceTypeProvider   *InstanceTypeProvider
 }
 
+// AWSConstraints are AWS specific constraints
+type AWSConstraints struct {
```

Comment:
I like this concept. Could we make a new file to hold this struct?

Reply:
Another alternative
```
func GetCapacityType(cloudprovider.Constraints) string {
 ...
}
```
IIUC, you only really need to call this once, and it simplifies lots of code changes. This struct doesn't fundamentally hold additional data, it's just formed differently. This makes me feel like we probably just need a new window (i.e. a function) onto the underlying struct.

Reply:
Yeah!

Reply:
It needs to be called in InstanceTypeProvider too.  I could do a hybrid where AWSConstraints is a struct that embeds a pointer to `cloudprovider.Constraints` and put a method on that struct of `GetCapacityType()`. That pattern will make it easier to add other AWS specific constraints like the launch template overrides coming soon.

awsconstraints.go
```
type AWSConstraints struct {
	*cloudprovider.Constraints
}

func NewAWSConstraints(constraints *cloudprovider.Constraints) *AWSConstraints {
	return &AWSConstraints{
		constraints,
	}
}

func (a *AWSConstraints) GetCapacityType() string {
	capacityType, ok := a.Labels[capacityTypeLabel]
	if !ok {
		capacityType = capacityTypeOnDemand
	}
	return capacityType
}
```